src/sth_database.js

```diff
@@ -169,52 +171,79 @@
    * @return {string} The database name
    */
   function getDatabase(service) {
-    return sthConfig.DB_PREFIX + '_' + service;
+    return sthConfig.DB_PREFIX + service;
   }
 
   /**
    * Return the name of the collection which will store the raw events
+   * @param {string} databaseName The database name
    * @param {string} servicePath The service path of the entity the event is related to
    * @param {string} entityId The entity id related to the event
    * @param {string} entityType The type of entity related to the event
    * @param {string} attrName The attribute id related to the event
    * @returns {string} The collection name
    */
-  function getCollectionName4Events(servicePath, entityId, entityType, attrName) {
+  function getCollectionName4Events(databaseName, servicePath, entityId, entityType, attrName) {
+    var collectionName4Events;
     switch(sthConfig.DATA_MODEL) {
       case sthConfig.DATA_MODELS.COLLECTIONS_PER_SERVICE_PATH:
-        return sthConfig.COLLECTION_PREFIX + '_' + servicePath;
+        collectionName4Events = servicePath;
+        break;
       case sthConfig.DATA_MODELS.COLLECTIONS_PER_ENTITY:
-        return sthConfig.COLLECTION_PREFIX + '_' + servicePath + '_' + entityId + (entityType ? '_' + entityType : '');
+        collectionName4Events =  servicePath + '_' + entityId + (entityType ? '_' + entityType : '');
+        break;
       case sthConfig.DATA_MODELS.COLLECTIONS_PER_ATTRIBUTE:
-        return sthConfig.COLLECTION_PREFIX + '_' + servicePath + '_' + entityId + (entityType ? '_' + entityType : '') +
+        collectionName4Events =  servicePath + '_' + entityId + (entityType ? '_' + entityType : '') +
           '_' + attrName;
+        break;
     }
+    // The maximum number of bytes accepted by MongoDB for namespaces is 120 bytes.
+    var limit = 119 - bytesCounter.count(databaseName) - bytesCounter.count(sthConfig.COLLECTION_PREFIX) -
+      bytesCounter.count('.aggr');
```

Comment:
Despite this is a method for obtaining the collection name for raw data, I guess it is necessary to have into account the number of bytes for ".aggr" since the hash-based collection name must match both for raw and aggregated, right?